nautobot/core/ui/object_detail.py

```diff
@@ -378,25 +384,111 @@ def __init__(
+        related_field_name = self.related_field_name or obj._meta.model_name
+        return_url = context["return_url"]
+
+        if self.add_button_route == "default":
+            body_content_table = context[self.table_key]
+
+            body_content_table_model = body_content_table.Meta.model
+            permission_name = get_permission_for_model(body_content_table_model, "add")
+            if request.user.has_perms([permission_name]):
+                with contextlib.suppress(NoReverseMatch):
+                    add_route = reverse(get_route_for_model(body_content_table_model, "add"))
+                    body_content_table_add_url = f"{add_route}?{related_field_name}={obj.pk}&return_url={return_url}"
+
+        elif self.add_button_route not in [None, "default"]:
+            if request.user.has_perms(self.add_permissions or []):
+                add_route = reverse(self.add_button_route)
+                body_content_table_add_url = f"{add_route}?{related_field_name}={obj.pk}&return_url={return_url}"
+
+        return body_content_table_add_url
 
     def get_extra_context(self, context):
         """Take the context value under `self.table_key` and add it to the context under key `"body_content_table"`."""
-        return {"body_content_table": context[self.table_key]}
+        body_content_table = context[self.table_key]
+        request = context["request"]
+
+        if self.exclude_fields is not None:
+            body_content_table.exclude = self.exclude_fields
+        elif self.include_fields is not None:
+            body_content_table.Meta.default_columns = self.include_fields
+
+        per_page = self.max_display_count if self.max_display_count is not None else get_paginate_count(request)
+        paginate = {"paginator_class": EnhancedPaginator, "per_page": per_page}
+        RequestConfig(request, paginate).configure(body_content_table)
+        more_queryset_count = body_content_table.data.data.count() - per_page
+
+        obj = context["obj"]
+        body_content_table_model = body_content_table.Meta.model
+        related_field_name = self.related_field_name or obj._meta.model_name
+
+        try:
+            list_route = reverse(get_route_for_model(body_content_table_model, "list"))
+            body_content_table_list_url = f"{list_route}?{related_field_name}={obj.pk}"
+        except NoReverseMatch:
+            body_content_table_list_url = None
+
+        body_content_table_add_url = self._get_table_add_url(context)
+
+        return {
+            "body_content_table": body_content_table,
+            "body_content_table_add_url": body_content_table_add_url,
+            "body_content_table_list_url": body_content_table_list_url,
+            "body_content_table_verbose_name": body_content_table_model._meta.verbose_name,
+            "body_content_table_verbose_name_plural": body_content_table_model._meta.verbose_name_plural,
+            "more_queryset_count": max(more_queryset_count, 0),
```

Comment:
Maybe move the `max()` call up to line 470 where `more_queryset_count` is set in the first place?